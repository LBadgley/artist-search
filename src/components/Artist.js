import React from 'react';
import PropTypes from 'prop-types';

function Artist({ name }) {
  return <span>Name: {name}</span>;
}

Artist.propTypes = {
  name: PropTypes.string.isRequired
};

export default Artist;
