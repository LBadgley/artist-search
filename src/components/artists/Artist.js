import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

function Artist({ name, id }) {
  return <Link to={`/releases/${name}/${id}`}>{name}</Link>;
}

Artist.propTypes = {
  name: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired
};

export default Artist;
