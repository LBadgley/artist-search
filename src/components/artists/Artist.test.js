import React from 'react';
import { shallow } from 'enzyme';
import Artist from './Artist';

describe('artist test', () => {
  it('renders the artist', () => {
    const wrapper = shallow(<Artist />);
    expect(wrapper).toMatchSnapshot();
  });
});
