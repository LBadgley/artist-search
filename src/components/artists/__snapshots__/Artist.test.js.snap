// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`artist test renders the artist 1`] = `
<Link
  to="/releases/undefined/undefined"
/>
`;
